src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -2257,13 +2249,7 @@ private bool NeedsTypeLayoutCheck(TypeDesc type)
 
         private bool HasLayoutMetadata(TypeDesc type)
         {
-            if (type.IsValueType && (MarshalUtils.IsBlittableType(type) || ReadyToRunMetadataFieldLayoutAlgorithm.IsManagedSequentialType(type)))
-            {
-                // Sequential layout
-                return true;
-            }
-
-            return false;
+            return MetadataFieldLayoutAlgorithm.IsBlittableOrManagedSequentialType(type) || (type is MetadataType mt && mt.IsExplicitLayout);
```

Comment:
I would simplify this to `mt.IsExplicitLayout || mt.IsSequentialLayout`. Maybe even inline it at the callsite.